### PR TITLE
Create empty as

### PR DESCRIPTION
**Pull request by guykoh**

### CODE REVIEW

File: dbt/adapters/clickhouse/impl.py [hunk 1]
```diff
@@ -76,6 +78,15 @@ def get_clickhouse_cluster_name(self):
         if conn.credentials.cluster:
             return '"{}"'.format(conn.credentials.cluster)
 
+    @available
+    def get_clickhouse_version(self):
+        conn = self.connections.get_if_exists()
+        if isinstance(conn.handle, HttpClient):
+            return conn.handle.server_version
+        elif isinstance(conn.handle, ChNativeAdapter):
```
Comment (genzgd) [inline dbt/adapters/clickhouse/impl.py:86]: Can you abstract this away into the ChNativeAdapter with a `server_version` @property field?

File: dbt/include/clickhouse/macros/adapters.sql [hunk 2]
```diff
@@ -66,27 +66,44 @@
...
+        {{ partition_cols(label="partition by") }}
+    {%- else %}
+        create table {{ relation.include(database=False) }}
+        {{ on_cluster_clause(label="on cluster") }}
+        {{ engine_clause(label="engine") }}
+        {{ order_cols(label="order by") }}
+        {{ primary_key_clause(label="primary key") }}
+        {{ partition_cols(label="partition by") }}
+        {%- set ch_version = adapter.get_clickhouse_version() -%}
+        {% if ch_version > '22.7.1' -%}
```
Comment (genzgd) [inline dbt/include/clickhouse/macros/adapters.sql:100]: same problem with string comparison

File: dbt/include/clickhouse/macros/adapters.sql [hunk 3]
```diff
@@ -107,6 +124,13 @@
   {{ return(load_result('list_schemas').table) }}
 {% endmacro %}
 
+{% macro clickhouse__get_version() %}
```
Comment (genzgd) [inline dbt/include/clickhouse/macros/adapters.sql:127]: Why do we need this macro?

File: dbt/include/clickhouse/macros/adapters.sql [hunk 4]
```diff
@@ -66,27 +66,44 @@
...
 
 {% macro clickhouse__create_table_as(temporary, relation, sql) -%}
-  {%- set sql_header = config.get('sql_header', none) -%}
+    {% set create_table = create_table_or_empty(temporary, relation, sql) %}
+    {%- set ch_version = adapter.get_clickhouse_version() -%}
+    {% if ch_version < '22.7.1' -%}
+        {{ create_table }}
+        {{ adapter.get_model_settings(model) }}
+    {%- else %}
+        {% call statement('creat_table_empty') %}
```
Comment (genzgd) [inline dbt/include/clickhouse/macros/adapters.sql:75]: `creat` typo?

**Review by genzgd — CHANGES_REQUESTED** [review]

File: dbt/include/clickhouse/macros/adapters.sql (hunk 3 above)
Comment (guykoh) [reply dbt/include/clickhouse/macros/adapters.sql]: Forgot to remove it - removed now.

**Review by genzgd — APPROVED** [review]

**Pull request merged**